src/sknnr_spatial/datasets/_base.py

```diff
@@ -54,10 +54,13 @@ def _load_rasters_to_dataset(
 def _load_rasters_to_array(file_paths: list[Path]) -> NDArray:
     """Load a list of rasters as a numpy array."""
     arr = None
-    for path in file_paths:
+    for i, path in enumerate(file_paths):
         with rasterio.open(path) as src:
             band = src.read(1)
-            arr = band if arr is None else np.dstack((arr, band))
+            if arr is None:
+                arr = np.empty((len(file_paths), *band.shape), dtype=band.dtype)
```

Comment:
Yeah, you're absolutely right! It looks like this will down-cast subsequent `band` arrays to make them fit in the `arr` dtype, so floats would get rounded. That would've been a tough bug to track down!

I rewrote this closer to the original but using `concatenate` instead of `dstack`, and added a test to make sure that loading an `int` and `float` raster gives you a `float` raster back.

Reply:
Perfect!  And of course you went the extra step and wrote an awesome test for this as well!